Module.php

```diff
@@ -185,6 +214,156 @@ public function handleItemApiHydratePre(Event $event)
         $request->setContent($data);
     }
 
+    public function addUserFormElement(Event $event): void
+    {
+        /** @var \Omeka\Form\UserForm $form */
+        $form = $event->getTarget();
+        if ($form->getOption('is_public')) {
+            return;
+        }
+
+        /**
+         * @var \Omeka\Entity\User $user
+         */
+        $services = $this->getServiceLocator();
+        $settings = $services->get('Omeka\Settings');
+        $entityManager = $services->get('Omeka\EntityManager');
+
+        // The user is not the current user, but the user in the form.
+        // It may be empty for a new user.
+        $userId = $services->get('Application')->getMvcEvent()->getRouteMatch()->getParam('id');
+        $user = $userId ? $entityManager->find(\Omeka\Entity\User::class, $userId) : null;
+
+        $mainDirectory = $settings->get('file_sideload_directory', '');
+        $this->directory = $mainDirectory;
+        $this->deleteFile = $settings->get('file_sideload_delete_file') === 'yes';
+
+        // Manage a direct creation (no id).
+        if ($user) {
+            /** @var \Omeka\Settings\UserSettings $userSettings */
+            $userSettings = $services->get('Omeka\Settings\User');
+            $userSettings->setTargetId($userId);
+            $userDir = $userSettings->get('filesideload_user_dir', '');
+        } else {
+            $userDir = '';
+        }
+
+        // Max 2 levels because it is the most common case: one level for the
+        // user or institution, and when the first is the institution, a second
+        // for the collections or the users. Maybe 3, but no more for now in
+        // order to avoid issues with big directories.
+        // This option only applies to the user interface anyway.
+        $maxDepth = 2;
+        $maxDirs = (int) $settings->get('file_sideload_max_directories');
+        $directories = $mainDirectory ? $this->listDirs($mainDirectory, $maxDepth, $maxDirs) : [];
+
+        $fieldset = $form->get('user-settings');
+        $fieldset
+            ->add([
+                'name' => 'filesideload_user_dir',
+                'type' => \Laminas\Form\Element\Select::class,
+                'options' => [
+                    'label' => 'Server directory', // @translate
```

Comment:
"Server directory" as a label without explanation that it pertains to this module would be confusing, no?

Reply:
This is not related to this pr.

Reply:
I renamed the label.